be/src/util/thread.cpp

```diff
@@ -169,6 +174,81 @@ void ThreadMgr::remove_thread(const pthread_t& pthread_id, const std::string& ca
     ANNOTATE_IGNORE_READS_AND_WRITES_END();
 }
 
+void ThreadMgr::start_instrumentation(const WebPageHandler::ArgumentMap& args, EasyJson* ej) const {
+    const auto* category_name = FindOrNull(args, "group");
+    if (category_name) {
+        bool requested_all = (*category_name == "all");
+        ej->Set("requested_thread_group", EasyJson::kObject);
+        (*ej)["group_name"] = escape_for_html_to_string(*category_name);
+        (*ej)["requested_all"] = requested_all;
+
+        // The critical section is as short as possible so as to minimize the delay
+        // imposed on new threads that acquire the lock in write mode.
+        vector<ThreadDescriptor> descriptors_to_print;
+        if (!requested_all) {
+            MutexLock l(&_lock);
+            const auto* category = FindOrNull(_thread_categories, *category_name);
+            if (category) {
+                for (const auto& elem : *category) {
+                    descriptors_to_print.emplace_back(elem.second);
+                }
+            }
+        } else {
+            MutexLock l(&_lock);
+            for (const auto& category : _thread_categories) {
+                for (const auto& elem : category.second) {
+                    descriptors_to_print.emplace_back(elem.second);
+                }
+            }
+        }
+
+        EasyJson found = (*ej).Set("found", EasyJson::kObject);
```

Comment:
If category is not found, we will also output normally?

Reply:
I changed it, if category is not found, we will not show it